presto-orc/src/test/java/com/facebook/presto/orc/TestOrcBloomFilters.java

```diff
@@ -0,0 +1,333 @@
+    }
+
+    @Test
+    public void testBloomFilterPredicateValuesNonExisting()
+            throws Exception
+    {
+        ImmutableMap<Object, Type> testValues = getTestElements();
+
+        BloomFilter bloomFilter = new BloomFilter(testValues.size() * 10, 0.01D);
+
+        for (Map.Entry<Object, Type> testValue : testValues.entrySet()) {
+            boolean matched = TupleDomainOrcPredicate.checkInBloomFilter(bloomFilter, testValue.getKey(), testValue.getValue());
+            assertFalse(matched, "type " + testValue.getKey().getClass());
+        }
+
+        // test unsupported type
+        assertTrue(TupleDomainOrcPredicate.checkInBloomFilter(bloomFilter, new Date(), null), "unsupported type Date should always return true");
+    }
+
+    @Test
+    public void testExtractValuesFromSingleDomain()
+            throws Exception
+    {
+        ImmutableMap.Builder<Type, Object> testBuilder = ImmutableMap.builder();
+        testBuilder.put(BOOLEAN, true);
+        testBuilder.put(INTEGER, 1234L); // internally treated as long
+        testBuilder.put(BIGINT, 4321L);
+        testBuilder.put(DOUBLE, 0.123D);
+        BlockBuilder blockBuilder = VARCHAR.createBlockBuilder(new BlockBuilderStatus(), 15);
+        VARCHAR.writeString(blockBuilder, TEST_STRING);
+        Block block = blockBuilder.build();
+        testBuilder.put(VARCHAR, VARCHAR.getSlice(block, 0));
+        ImmutableMap<Type, Object> testValues = testBuilder.build();
+
+        for (Map.Entry<Type, Object> testValue : testValues.entrySet()) {
+            Domain predicateDomain = Domain.singleValue(testValue.getKey(), testValue.getValue());
+            Collection<Object> objects = TupleDomainOrcPredicate.predicateValuesFromDomain(predicateDomain);
+            assertEquals(objects.size(), 1);
+            assertEquals(objects.iterator().next(), testValue.getValue());
+        }
+    }
+
+    @Test
+    // simulate query on a 2 columns where 1 is used as part of the where, with and without bloom filter
+    public void testMatches()
+            throws Exception
+    {
+        // stripe columns
+        TestingColumnHandle testingColumnHandle = new TestingColumnHandle("bigint_0");
+        Domain testingColumnHandleDomain = Domain.singleValue(BIGINT, 1234L);
+        TupleDomain.ColumnDomain<TestingColumnHandle> column = new TupleDomain.ColumnDomain<>(testingColumnHandle, testingColumnHandleDomain);
+
+        TestingColumnHandle testingColumnHandle1 = new TestingColumnHandle("bigint_1");
+        Domain testingColumnHandleDomain1 = Domain.singleValue(BIGINT, 234L);
+        TupleDomain.ColumnDomain<TestingColumnHandle> column1 = new TupleDomain.ColumnDomain<>(testingColumnHandle1, testingColumnHandleDomain1);
+
+        // predicate consist of the bigint_0 = 1234
+        TupleDomain<TestingColumnHandle> effectivePredicate = TupleDomain.fromColumnDomains(Optional.<List<TupleDomain.ColumnDomain<TestingColumnHandle>>>of(ImmutableList.of(column)));
```

Comment:
No it's meant to simulate a table with 2 columns which are both referenced but only 1 is part of the predicate.